Mail/Views/Bottom sheets/Actions/ActionsView.swift

```diff
@@ -203,3 +211,37 @@ struct ActionButtonLabel: View {
         .padding(value: .medium)
     }
 }
+
+private struct PerformActionRequest {
+    let targetMessages: [Message]
+    let action: Action
+    let origin: ActionOrigin
+    let isMultipleSelection: Bool
+    let completionHandler: ((Action) -> Void)?
+}
+
+private func performActionAndTrack(
+    request: PerformActionRequest,
+    actionsManager: ActionsManager,
+    dismiss: @escaping () -> Void
+) {
+    @InjectService var matomo: MatomoUtils
+    dismiss()
+    Task {
+        await tryOrDisplayError {
+            try await actionsManager.performAction(
+                target: request.targetMessages,
+                action: request.action,
+                origin: request.origin
+            )
+            request.completionHandler?(request.action)
+
+            matomo.trackAction(
+                action: request.action,
+                origin: request.origin,
+                numberOfItems: request.targetMessages.count,
+                isMultipleSelection: request.isMultipleSelection
+            )
+        }
+    }
+}
```

Comment:
Not a big fan of this struct. Even if Sonar doesn't agree, having a function with 5 parameters makes more sense.

Having a top level function like `performActionAndTrack` is not a good practice.